loadtest/store/memstore/random.go

```diff
@@ -59,45 +48,32 @@ func (s *MemStore) RandomEmoji() (model.Emoji, error) {
 
 // RandomChannelMember returns a random channel member for a channel.
 func (s *MemStore) RandomChannelMember(channelId string) (model.ChannelMember, error) {
-	genericMap := make(map[string]interface{})
+	var chanMemberMap map[string]*model.ChannelMember
 	for k, v := range s.channelMembers {
 		if k == channelId {
-			for k1, v1 := range v {
-				genericMap[k1] = v1
-			}
+			chanMemberMap = v
 			break
 		}
 	}
-	cm := selectRandomFromMap(genericMap)
-	return *(cm.(*model.ChannelMember)), nil
+	key := pickRandomKeyFromMap(chanMemberMap).(string)
+	return *chanMemberMap[key], nil
 }
 
 // RandomTeamMember returns a random team member for a team.
 func (s *MemStore) RandomTeamMember(teamId string) (model.TeamMember, error) {
-	genericMap := make(map[string]interface{})
+	var teamMemberMap map[string]*model.TeamMember
 	for k, v := range s.teamMembers {
 		if k == teamId {
-			for k1, v1 := range v {
-				genericMap[k1] = v1
-			}
+			teamMemberMap = v
 			break
 		}
 	}
-	tm := selectRandomFromMap(genericMap)
-	return *(tm.(*model.TeamMember)), nil
+	key := pickRandomKeyFromMap(teamMemberMap).(string)
+	return *teamMemberMap[key], nil
 }
 
-// selectRandomFromMap flattens a map to a slice
-// and then returns a random index from the slice.
-// We do this, because even though the map iteration order
-// is "undefined" in the spec, practically it returns the
-// same set of values always.
-func selectRandomFromMap(m map[string]interface{}) interface{} {
-	items := make([]interface{}, len(m))
-	i := 0
-	for _, val := range m {
-		items[i] = val
-		i++
-	}
-	return items[rand.Intn(len(items))]
+func pickRandomKeyFromMap(m interface{}) interface{} {
+	keys := reflect.ValueOf(m).MapKeys()
+	idx := rand.Intn(len(keys))
```

Comment:
`rand.Intn` will panic if given 0 as input. So I propose we either return an error or we make 100% sure we always call this with a non-empty map. Also a test case for this would be nice.

Reply:
Good catch.